### PR TITLE
KOGITO-8611 Added Fix for Copy/Cut/Paste Shortcuts not working in Mac

### PR DESCRIPTION
**Pull request by skannanath**
Hi Team,

This pull request addresses an issue where the standard copy, cut, and paste keyboard shortcuts (⌘C, ⌘X, ⌘V) don't work within the DMN Editor extension for VSCode on macOS.

The changes in this PR resolve this functionality gap, ensuring a consistent user experience for clipboard interactions within the DMN editor.

**Following is the summary of changes done in this PR:**
1. Enables Shortcut binds (⌘C, ⌘X, ⌘V) for cut, copy, and paste actions.
2. Enables selected content to be cut, copy or pasted.
3. Supports copying selected text, including content that spans multiple lines. This addresses the current issue where only text before newlines is copied.

Manual testing confirmed successful clipboard functionality after the changes. 

ezgif-7-0cd2b36f07

Please review this pull request and provide any feedback.

More details on the issue can be found in this Jira Ticket. https://issues.redhat.com/browse/KOGITO-8611

Regards,
Kannan S

### CODE REVIEW

**Comment by yesamer** [comment]
@skannanath Thank you for your contribution!! 

/cc @ljmotta @tiagobento @jomarko

**Comment by ljmotta** [comment]
@skannanath Thanks for your contribution. The 0.26.0 is a quite old version. I've tried the shortcuts on `main` branch on a M1 Mac machine without any problems. Could you please try it too?

To generate a VSIX of the main branch, please clone the repo, and on the monorepo root:

```
pnpm bootstrap -F dmn-vscode-extension...
pnpm -F dmn-vscode-extension^... build:dev
pnpm -F dmn-vscode-extension build:prod
```

A VSIX will be generated on the `packages/dmn-vscode-extension/dist` folder.


https://github.com/apache/incubator-kie-tools/assets/24302289/81db80b1-5350-45bc-89b3-a41394098bb0

**Comment by skannanath** [comment]
Hi @ljmotta,

That's good news! I was able to build the VSIX extension and test the clipboard functionality. You're right, the issue is not present in the FEEL Inputs. However, the clipboard issue appears to be still present in the New DRD View, where copy-paste functionality is still missing, just like in the previous version of DRD View(Not fixed in this PR).

Do you have an estimated timeline for when this new editor version might be rolled out? In the meantime, is it possible for us to try out the new version and provide feedback?

Also, are there any workarounds or recommendations we can follow until the new version is available?

Thanks!

**Comment by ljmotta** [comment]
@skannanath Great! Regarding the `New DRD View`, could you please give a bit more context? It would be great if you could create an issue on the [kie-issues](https://github.com/apache/incubator-kie-issues/) detailing the problem.

The new editor will be available on our next release. We are still working on some rough edges regarding it, but it should land soon. We post news every Friday regarding our meetings on the [kie-issues](https://github.com/apache/incubator-kie-issues/discussions) repository.

For the Sandbox, we have a [daily-dev](https://daily-dev-kie-sandbox-kie-sandbox.rhba-0ad6762cc85bcef5745bb684498c2436-0000.us-south.containers.appdomain.cloud/#/) environment which is updated every day with the `main` branch, it's important to notice that we do not recommend to be used in a production environment. Unfortunately, we don't have a similar for the VS Code extension, it would require to build manually as you did.

**Comment by skannanath** [comment]
Hi @ljmotta, Created an issue on apache/incubator-kie-issues (https://github.com/apache/incubator-kie-issues/issues/1164) detailing the DRD View clipboard issue. Additional information and steps to reproduce are included to assist with resolution.

**Comment by skannanath** [comment]
Hi @ljmotta, Our team is excited about the upcoming DMN editor version. As part of our planning process, we'd like to confirm backward compatibility with existing DMN 1.2 models. We currently utilize DMN 1.2 extensively, and due to the ongoing development of the DMN 1.5 specification, we intend to migrate to the new standard upon its official release.

image

Our primary concern is the behavior of the new editor with DMN 1.2 models. Specifically, we'd like to confirm if the editor will:

- Seamlessly open, edit, and save DMN 1.2 files without altering their functionality.
- Maintain compatibility with Drools versions 7 and 8 for running these models.

Our current observation suggests that the editor might convert 1.2 models to 1.5 during open and save with new editor. To ensure a smooth transition for our existing projects, we'd appreciate clarification on whether this behavior will be modified in the final release. Information on this aspect will be crucial for our planning.

Thank you for your time and assistance.

**Comment by yesamer** [comment]
@skannanath 
Going straight to your questions:

> Seamlessly open, edit, and save DMN 1.2 files without altering their functionality.

Not exactly. The new DMN Editor can open any DMN file version, but once loaded in the editor, it will be automatically converted to a 1.5 DMN version file. But your business logic will be preserved (I'll elaborate better later).

> Maintain compatibility with Drools versions 7 and 8 for running these models.

Drools 8 engine supports up to DMN 1.4, so, according to my response to the previous point, if you create or edit a DMN file with the new editor, that will be saved as 1.5 DMN version that IS NOT compatible with Drools 8 (and 7).

To give you more context:
Since we moved the project under the Apache group, We aim to support the latest DMN version as fast as we can, in both the engine and the editor side simultaneously. That wasn't the case in the past. With Drools 8, there was a discrepancy between the Engine and the Editor (the engine supported up to 1.4 version while the editor up to 1.2). 
With the next Apache KIE release, we are going to fully support 1.5.
I want to clarify that DMN 1.5 is no longer in BETA, so you can consider it as a stable version. Honestly, I don't know why the OMG site didn't still update that. 
DMN 1.5 specification guarantees backward compatibility with the previous DMN version, so your logic will be preserved when updating your DMN assets. 
I hope I give you all the information you need to make the best decision for your team and organization.
To recap, if you prefer to continue using the Drools 8 / Drools 7 engine, you must continue using the current released editor with DMN 1.2.
The moment you decide to use the upcoming Drools 10, you need to be aware that we are targeting DMN 1.5.

**Comment by yesamer** [comment]
DMN 1.5 approved by OMG architectural board 20.3.2023. [https://github.com/omg-dmn-taskforce/omg-dmn-](https://github.com/omg-dmn-taskforce/omg-dmn-spec/releases)

**Comment by jomarko** [comment]
What is the status of this Pull request please? Is it something we still incorporate a feedback in? Or is it something outdated now?

**Comment by skannanath** [comment]
> What is the status of this Pull request please? Is it something we still incorporate a feedback in? Or is it something outdated now?

@jomarko The fix addressed in this pull request has already been fixed in the master branch. Therefore, I'm closing this PR. Thanks!

@yesamer
